src/test/java/edu/cornell/kfs/pdp/batch/service/impl/CuExtractPaymentServiceImplTest.java

```diff
@@ -0,0 +1,188 @@
+    
+    
+    
+    private CuExtractPaymentServiceImpl cuExtractPaymentServiceImpl;
+
+    @Before
+    public void setUp() throws Exception {
+        cuExtractPaymentServiceImpl = new TestCuExtractPaymentServiceImpl();
+        File testOutputDirectory = new File(ACH_EXTRACT_TEST_FILE_PATH);
+        testOutputDirectory.mkdir();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        cuExtractPaymentServiceImpl = null;
+        FileUtils.forceDelete(new File(ACH_EXTRACT_TEST_FILE_PATH).getAbsoluteFile());
+    }
+
+    @Test
+    public void testCalculateHeaderDate() {
+        Date now = new Date(Calendar.getInstance().getTimeInMillis());
+        Date functionReturnedDate = cuExtractPaymentServiceImpl.calculateHeaderDate(now);
+        Calendar calendarCalculator = Calendar.getInstance();
+        calendarCalculator.setTimeInMillis(now.getTime());
+        calendarCalculator.add(Calendar.DATE, 1);
+        Date expectedDate = new Date(calendarCalculator.getTimeInMillis());
+        assertEquals("headerDate from function should equal date we calculate in the same manner:  ", expectedDate, functionReturnedDate);
+    }
+    
+    @Test
+    public void testAchFileHeaderDateIsOneDayAfterProcessDate() {
+        cuExtractPaymentServiceImpl.setAchBundlerHelperService(new TestAchBundlerHelperService());
+        
+        PaymentStatus extractedStatus = new PaymentStatus();
+        Date processDate = new Date(Calendar.getInstance().getTimeInMillis());
+        List<String> notificationEmailAddresses = new ArrayList<String>(Arrays.asList(NOTIFICATION_EMAIL_ADDRESSES));
+        
+        cuExtractPaymentServiceImpl.writeExtractAchFileMellonBankFastTrack(extractedStatus, ACH_EXTRACT_TEST_PATH_AND_FILE, processDate, yyyyMMddHHmmss_DATE_FORMATTER, notificationEmailAddresses);
+        
+        Date expectedHeaderDate = cuExtractPaymentServiceImpl.calculateHeaderDate(processDate);
+        assertTrue("ACH Extract Payments File Header Date is not what was expected.", isDateInOutputFileWhereExpected(HEADER_LINE_IN_OUTPUT_FILE, yyyyMMddHHmmss_DATE_FORMATTER.format(expectedHeaderDate)));
+        assertTrue("ACH Extract Payments File Payee Date is not what was expected.", isDateInOutputFileWhereExpected(FIRST_PAYEE_LINE_IN_OUTPUT_FILE, yyyyMMdd_DATE_FORMATTER.format(processDate)));
+    }
+    
+    private boolean isDateInOutputFileWhereExpected(int numberOfLinesToRead, String formattedDateExpected) {
+        BufferedReader inputReader = null;
+        boolean expectedWasFound = false;
+        try {
+            String lineWeWant = null;
+            inputReader = new BufferedReader(new FileReader(new String (ACH_EXTRACT_TEST_PATH_AND_FILE_FOR_VALIDATION)));
+            for (int i=0; i<numberOfLinesToRead; i++) {
+                lineWeWant = inputReader.readLine();
+            }
+            expectedWasFound = lineWeWant.contains(formattedDateExpected);
+        }
+        catch (IOException ie) {
```

Comment:
Not necessarily something I would make you change, but something I wanted to point out: The simplest thing for a unit test would be to just throw the IOException from this method and not bother with the try/catch. If there is an IOException, the test will still fail, and you let the testing framework do the work for you. The only time I catch Exceptions in unit tests is when I need to do processing after the point in which the exception occurs. Otherwise I just throw them, or use annotations to tell JUnit to expect them.